### DISCUSSION TITLE
Sonoff NS Panel + Esphome + termostato

### DISCUSSION THREAD

**Discussion by zanna1972** [indomus/forum#2347] [GENERALE - Discussioni generiche]
Ciao, ho acquistato un Sonoff NS Panel che voglio utilizzare come termostato in casa. Ho collegato uno dei due relè alla testina di riscaldamento e tutto funziona perfettamente usando la app eWelink. Ho provato poi ad integrarlo in Home Assistant prima usando Sonoff LAN, poi usando l'add-on ufficiale eWelink-ha-add-on, ma entrambi funzionano bene se lo si usa per pilotare delle lampadine, ma come termostato sono carenti e la richiesta di correzioni è difficile e lentissima. 
Ho quindi pensato di installare Esphome nella speranza di poterlo gestire completamente anche  per la modalità di termostato da Home Assistant. Ci sono diversi video su Youtube,  ma una bella guida step by step in italiano non l'ho trovata.
Se qualcuno lo ha fatto o se sa dove trovare questa guida sarebbe molto apprezzato.

**Comment by f4b1ux** [comment]
Io l'ho fatto, seguendo le varie guide trovate su internet (nessuna in italiano). Il problema non è tanto mettere esphome, cosa abbastanza semplice, il problema è riprogrammare il display nextion per farlo parlare con esphome...

Ti consiglio di guardare questo [video](https://www.youtube.com/watch?v=p-AK4o5jOSI&t=37s) Sempre di questo tizio avevo trovato un progetto che consentiva di generalizzare un po' andando a definire i comandi dell'nspanel direttamente da HA, ma non ho approfondito l'argomento avendo fatto tutto io custom.
